third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -1767,22 +1771,134 @@ impl ModuleContext<'_> {
+                                        number.to_u16().unwrap(),
+                                    ))
+                                } else {
+                                    self.error(
+                                        fun_env.get_id_loc(),
+                                        format!("attribute `{}` must have a number value between 0 and 255", name),
+                                    )
+                                }
+                            } else {
+                                self.error(
+                                    fun_env.get_id_loc(),
+                                    format!("attribute `{}` must have a number value", name),
+                                )
+                            }
+                        },
+                        well_known::UNPACK_VARIANT => {
+                            if let AttributeValue::Value(_, Value::Number(number)) = attribute_value
+                            {
+                                if number.to_u16().is_some() {
+                                    result.push(FF::FunctionAttribute::UnpackVariant(
+                                        number.to_u16().unwrap(),
+                                    ))
+                                } else {
+                                    self.error(
+                                        fun_env.get_id_loc(),
+                                        format!("attribute `{}` must have a number value between 0 and 255", name),
+                                    )
+                                }
+                            } else {
+                                self.error(
+                                    fun_env.get_id_loc(),
+                                    format!("attribute `{}` must have a number value", name),
+                                )
+                            }
+                        },
+                        well_known::TEST_VARIANT => {
+                            if let AttributeValue::Value(_, Value::Number(number)) = attribute_value
+                            {
+                                if number.to_u16().is_some() {
+                                    result.push(FF::FunctionAttribute::TestVariant(
+                                        number.to_u16().unwrap(),
+                                    ))
+                                }
+                            } else {
+                                self.error(
+                                    fun_env.get_id_loc(),
+                                    format!("attribute `{}` must have a number value", name),
+                                )
+                            }
+                        },
+                        well_known::BORROW_NAME => {
+                            if let AttributeValue::Value(_, Value::Number(number)) = attribute_value
+                            {
+                                if number.to_u16().is_some() {
```

Comment:
This code feels a bit weird: we have a conversion to `u16` (range: 0-65535), but we complain that if that conversion fails, value should be between 0-255.